src/App.test.js

```diff
@@ -12,3 +19,292 @@ test('renders a reading with the text `Pokédex`', () => {
+  const { getByText } = render(
+    <MemoryRouter initialEntries={['/']}>
+      <App />
+    </MemoryRouter>,
+  );
+
+  expect(getByText('Encountered pokémons')).toBeInTheDocument();
+});
+
+function renderWithRouter(
+  ui,
+  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
+) {
+  return {
+    ...render(<Router history={history}>{ui}</Router>),
+    history,
+  };
+}
+
+describe('Render the pokedex with the pokemon', () => {
+  test('2 render one image', () => {
+    const { getAllByText } = renderWithRouter(<App />);
+
+    expect(getAllByText(/Average weight/i).length).toBe(1);
+  });
+});
+
+describe('3 Next pokemon', () => {
+  afterEach(cleanup);
+  const namePokemon = ['Pikachu', 'Charmander', 'Caterpie', 'Ekans',
+    'Alakazam', 'Mew', 'Rapidash', 'Snorlax', 'Dragonair'];
+
+  test('3.1, 3.2 click on the button, appear next pokemon', () => {
+    const { getByText } = renderWithRouter(<App />);
+
+    const nextPokemon = getByText(/Próximo pokémon/i);
+    expect(nextPokemon).toBeInTheDocument();
+
+
+    for (let index = 0; index < namePokemon.length; index += 1) {
+      expect(getByText(namePokemon[index])).toBeInTheDocument();
+      fireEvent.click(nextPokemon);
+    }
+  });
+
+  test('3.3 Last pokemon to first pokemon', () => {
+    const { getByText } = renderWithRouter(<App />);
+
+    const nextPokemon = getByText(/Próximo pokémon/i);
+
+    for (let index = 0; index < namePokemon.length; index += 1) {
+      fireEvent.click(nextPokemon);
+    }
+    expect(getByText(/Pikachu/i)).toBeInTheDocument();
+  });
+});
+
+describe('4 The Pokédexx must have filter buttons', () => {
+  afterEach(cleanup);
+
+  test('4.1 ThePokedex filter the pokemons and show them', () => {
+    const { getByText, queryAllByText } = renderWithRouter(<App />);
+
+    const nextPokemon = getByText(/Próximo pokémon/i);
+    expect(nextPokemon).toBeInTheDocument();
+
+    pokemons.forEach((pokemon) => {
+      const buttonType = queryAllByText(pokemon.type)[1] || getByText(pokemon.type);
+      fireEvent.click(buttonType);
+      fireEvent.click(nextPokemon);
+      fireEvent.click(nextPokemon);
+      const samePokemon = queryAllByText(pokemon.type);
+      expect(samePokemon.length).toBe(2);
+      expect(pokemon.type).toBe(buttonType.innerHTML);
+    });
+  });
+});
+
+describe('5 reset filter', () => {
+  afterEach(cleanup);
+  const namePokemon = ['Pikachu', 'Charmander', 'Caterpie', 'Ekans',
+    'Alakazam', 'Mew', 'Rapidash', 'Snorlax', 'Dragonair'];
+
+  test('5.1 The text of button must be All', () => {
+    const { getByText } = renderWithRouter(<App />);
+    expect(getByText(/All/i)).toBeInTheDocument();
+  });
+
+  test('5.2 after clicking, The pokédex must show all pokemons', () => {
+    const { getByText } = renderWithRouter(<App />);
+
+    fireEvent.click(getByText(/All/i));
+
+    const firstTagP = document.getElementsByTagName('p')[0];
+    const nextPokemon = getByText(/Próximo pokémon/i);
```

Comment:
Nesses casos de utilização de mais uma palavra, o ideal é usar queryByText.
```suggestion
    const nextPokemon = queryByText(/Próximo pokémon/i);
```